src/indexer/log_merge_policy.rs

```diff
@@ -82,7 +99,12 @@ impl MergePolicy for LogMergePolicy {
 
         levels
             .iter()
-            .filter(|level| level.len() >= self.min_num_segments)
+            .filter(|level| {
+                level.len() >= self.min_num_segments
+                    || level
+                        .iter()
+                        .any(|segment| deletes_pct(segment) > self.max_del_docs_pct)
```

Comment:
Is it intended that we merge the whole level, if one segment is above the delete threshold?

I think it's better to assign them to variables first e.g.
```
let contains_segment_above_delete_threshold = level.iter().any(|segment| deletes_pct(segment) > self.max_del_docs_pct);
```

Reply:
I factored out a function `has_segment_above_deletes_threshold`